src/chttpd/src/chttpd_db.erl

```diff
@@ -1819,6 +1819,21 @@ db_attachment_req(#httpd{method = Method} = Req, Db, DocId, FileNameParts) when
         )
     ),
 
+    % Check if attachment exists
+    #doc_query_args{
+        rev=Rev,
+        options=Options
+    } = parse_doc_query(Req),
+    #doc{
+        atts=Atts
+    } = Doc = couch_doc_open(Db, DocId, Rev, Options),
+    case [A || A <- Atts, couch_att:fetch(name, A) == FileName] of
+        [] ->
+            throw({not_found, "Document is missing attachment"});
```

Comment:
This should be for `DELETE` only not `PUT`?

I saw the logic for opening the named attachment is similar to the code in `GET` handler, perhaps we can re-use it as `get_attachment(Req, Db, DocID, Rev, FileName) -> {ok,  Att} | {error, missing}` which both GET and DELETE can re-use?

Reply:
Fixed in https://github.com/apache/couchdb/pull/3667/commits/7190c7eb72ba4bdcb6112efa9aabe2436e094528.